com.unity.render-pipelines.universal/Documentation~/upgrade-guide-2022-1.md

```diff
@@ -0,0 +1,218 @@
+# Upgrading to version 2022.1 of the Universal Render Pipeline
+
+This page describes how to upgrade from an older version of the Universal Render Pipeline (URP) to version 2022.1.
+
+## Upgrading from URP 2021.2
+
+* An error will be issued when instances of `ScriptableRendererFeature` attempt to access render targets before they are allocated by the renderer.
+
+   The `ScriptableRendererFeature` has a new virtual function called `SetupRenderPasses` which is called when render targets are allocated and ready to be used.
+
+  If your code uses `renderer.cameraColorTarget` or `renderer.cameraDepthTarget` inside of the `AddRenderPasses` override, then that use needs to be moved to `SetupRenderPasses`.
```

Comment:
How does this affect passes added from e.g. components rather than renderer features?

Is there a point in the ScriptableRenderPass where the user can assume that  render targets are allocated?

Reply:
The issue with features was that universal renderer was calling `AddRenderPasses` before allocation, by design.
The targets are allocated right after in the `Setup` of the `ScriptableRenderer`.
If components are trying to access targets before the setup, then it's too early. Anytime after this is fine.